Separated News API config

Divy0409 opened the pull request:
## Using this Pull Request Template
1. In the Description of Changes section below, please fill out the details requested
2. Create the pull request
3. Check the boxes in the Type of Change section that apply to your changes
4. Check the boxes in the Checklist Before Review section that apply
5. Assign at least two reviewers to review your pull request by clicking the settings icon next to Reviewers

## Description of Changes
Please document the following about this pull request:
* I have created separate config option for news data. It works completely fine however there are some UI glitches those I'm unable to solve. 
### Identify the organization for which you completed this work
If this work is associated with a university course, please identify the university/course/semester (ex. MTU/MIS3200/Fall2023).
* List the organizaiton here: 
* MTU


### Identify what team you are on
* List the team here:
* GA/RA

### Identify the task, story, or GitHub issue associated with this work
* Identify the task/story/issue here:
* Create separate config option for News Data

### Identify the particular changes that you made for this pull request
* Describe the changes here: 
* I have created separate config option for news data. It works completely fine however there are some UI glitches those I'm unable to solve. 
* Moreover, to work with this branch you will have to add NEWS_API_KEY in your .env file.

## Type of Change (check all that apply)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation addition/update (non-breaking change that adds documentation to the system)

## Checklist Before Review
- [x] Do you want to keep your branch for future changes or delete it once it is reviewed?
- [x] I have followed the git practices outlined [here](GitPractices.md) to ensure code conflicts are minimized
- [x] My code follows the style guidelines of this project outlined [here](documentation/StyleGuide.md)
- [x] I have performed a self-review of my code. It runs without errors and follows the [Best Coding Practices](BestCodingPractices.md)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] There are no API keys, passwords, usernames, or other secrets in the code. I am using .env files for all secrets.
- [ ] I have added tests that prove my fix is effective or that my feature works

Divy0409 committed:
Separated News API config

jeffreywallphd committed:
small fixed bugs in API config and added Report and Ratio config area

jeffreywallphd committed:
Merge branch 'main' of https://github.com/jeffreywallphd/AutoProphet into API_Config

jeffreywallphd reviewed (commented):
@Divy0409 I fixed a few of the small issues you mentioned. I also added a Ratio and Report config area because it aligns with what I'm working on in the SECData branch. I separated the idea of Report and Ratio. Now we have configurations for Ratios for the ratios on the Stock & Fund page and Reports for reconstructing SEC 10-K and 10-Q reports. I ran everything and it seemed to work. Before I approve, I just want you to make sure that changes I made are not causing issues for you. After you're done testing, please let me know and I will approve and merge.

Divy0409 commented:
> @Divy0409 I fixed a few of the small issues you mentioned. I also added a Ratio and Report config area because it aligns with what I'm working on in the SECData branch. I separated the idea of Report and Ratio. Now we have configurations for Ratios for the ratios on the Stock & Fund page and Reports for reconstructing SEC 10-K and 10-Q reports. I ran everything and it seemed to work. Before I approve, I just want you to make sure that changes I made are not causing issues for you. After you're done testing, please let me know and I will approve and merge.

I have reviewed your changes and checked everything and everything is working fine. Thank you again for looking into this and adding all those extra configuration options.

jeffreywallphd reviewed (approved):
This has now been reviewed by myself and @Divy0409. Thanks for your efforts in setting up multiple configurations. It made it very easy to add new ones.
